### ISSUE TITLE
Debian repository

### ISSUE THREAD

**Issue by m3thm4th** [libredeb/lightpad#32]
It would be nice to have a debian repository to have automatic updates on Debian. This method could also replace the PPA on Ubuntu, adding security and expanding the user base of this application.

**Comment by tamburro92** [comment]
Added a request in debian, we need a maintainer:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1079219

**Comment by libredeb** [comment]
Let me boost this issue to see if we can achieve this goal. I would be honored to make it happen. Give me time

**Comment by fossfreedom** [comment]
@libredeb beginning to look at packaging this for debian.

Issue 1 - can I make a gentle suggestion please - move the debian folder to a separate branch.  For most projects the master/main branch is just the source code - distro specific capabilities such as the debian folder aren't normally in the source tree.  Understand though if you don't wish to do this - just a suggestion.

**Comment by fossfreedom** [comment]
@libredeb 

Would you like me to make separate issues?

Issue 2

Debian package

```
W: lightpad: appstream-metadata-validation-failed Problems reported by "appstreamcli validate-tree".
N: 
N:   The specified AppStream metadata file fail to validate using 'appstreamcli
N:   validate-tree --no-net path-to-package-root'.
N: 
N:   Please refer to https://wiki.debian.org/AppStream/Guidelines for details.
N: 
N:   Visibility: warning
N:   Show-Always: no
N:   Check: appstream-metadata
N:
``` 

indeed running the following produces the output

```
ppstreamcli validate data/com.github.libredeb.lightpad.appdata.xml 
I: com.github.libredeb.lightpad:9: description-first-para-too-short
     LightPad is a lightweight, look nice and powerful application launcher.
W: com.github.libredeb.lightpad:~: url-homepage-missing
E: com.github.libredeb.lightpad:~: desktop-app-launchable-missing
I: com.github.libredeb.lightpad:~: content-rating-missing
I: com.github.libredeb.lightpad:~: developer-info-missing

✘ Validation failed: errors: 1, warnings: 1, infos: 3, pedantic: 1
``

**Comment by lozanotux** [comment]
@fossfreedom yes, this is issue is more of a future goal.

We don't need to separate the issues, we can attack both issues in this same one.

For the **issue 1**, do you have an example of this at hand? I don't understand if the `debian` folder need to be alone in another branch or next by the all other files. But yes, I need to do that. Let me organize it.

For the **issue 2**, let me improve this, there are probably missing entries in the XML. I have a complete example of this.

**Comment by libredeb** [comment]
@fossfreedom yes, this is issue is more of a future goal.

We don't need to separate the issues, we can attack both issues in this same one.

For the **issue 1**, do you have an example of this at hand? I don't understand if the `debian` folder need to be alone in another branch or next by the all other files. But yes, I need to do that. Let me organize it.

For the **issue 2**, let me improve this, there are probably missing entries in the XML. I have a complete example of this.

**Comment by fossfreedom** [comment]
For example https://github.com/ubuntubudgie/budgie-extras

We hold the debian package on this branch

https://github.com/UbuntuBudgie/budgie-extras/tree/debian

**Comment by libredeb** [comment]
@fossfreedom I'm finished with both issues. Can you validate that it meets the expected standards?

**Comment by fossfreedom** [comment]
few more observations:

1. meson.build says "0.0.8" - the current version is 0.0.9
2. Looking at all the .vala files the copyright is GPL-2+ whereas COPYING is GPL-3 - can we please clarify - say  in the README - that there is a difference in licensing. If this is a mistake, please can the the vala copyright and COPYING be changed to the same license - e.g. GPL-2+ across everything?
3. Config.vala is missing a copyright statement

**Comment by libredeb** [comment]
@fossfreedom ooh, my bad... i'll fixing that as soon as possible. Thank you so much!

**Comment by libredeb** [comment]
@fossfreedom all done!

![Image](https://github.com/user-attachments/assets/49f54e31-902d-4b19-af9a-628fcdf63af6)

**Comment by fossfreedom** [comment]
Some distros like Debian and Fedora are encouraging maintainers to use GPG signed release tarballs - these are part of the assets for each release.  The key advantages are that you as a maintainer have confidence that the package in a distro is absolutely using your source code and not some variant.  The other advantage is similarly end-users have confidence in their distro package because its using code recommended by the project maintainer.

Don't worry if you don't want to-do this.  Really your choice.  But in future if you do want to use release tarballs one way is how we do things on Ubuntu Budgie and BuddiesofBudgie

https://github.com/BuddiesOfBudgie/budgie-desktop/blob/main/scripts/mkRelease.sh

Look at the release tarball - you will see the signed signature - the .asc file included with the assets. https://github.com/BuddiesOfBudgie/budgie-desktop/releases/tag/v10.9.2

$VTAR there is your gpg key as an environment variable.

https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key

On Ubuntu Budgie we use a variant of this release script https://github.com/UbuntuBudgie/budgie-extras/blob/master/mkrelease.sh - the git-archive-all.sh is from here https://github.com/fabacab/git-archive-all.sh

Your GPG key needs to be registered publicly

https://security.stackexchange.com/questions/181208/sending-the-gpg-key-to-keyserver

**Comment by libredeb** [comment]
@fossfreedom you are right, signing the code offers greater security and reliability in the released code. I just uploaded the signed release tarball.

**Replying to issue 2:**  good point, this is because when this project started it (in 2013) was not considered wayland yet, and libwnck (Window Navigator Construction Kit) was used to allow LightPad to do fullscreen on the default screen. This is now deprecated and also this library was only compatible with X11:

![Image](https://github.com/user-attachments/assets/1d3e548e-c550-4956-8d78-17f043e18b3c)

I've fixed this issue both in the code and in the packaging. 

Cheers!

**Comment by fossfreedom** [comment]
Did you upload your gpg to a public key server?  I can't see it

```
gpg --search libredeb@gmail.com
gpg: data source: https://keys.openpgp.org:443
gpg: key "libredeb@gmail.com" not found on keyserver
gpg: keyserver search failed: Not found
```

**Comment by libredeb** [comment]
@fossfreedom yes, I've upload it to ubuntu keyserver:

![Image](https://github.com/user-attachments/assets/32946799-acfd-4039-81e7-1339a1393b6f)

Is better to have in [openpgp.org](https://keys.openpgp.org) too?

**Comment by fossfreedom** [comment]
@libredeb cheers!

 did you sign the v0.0.9 tarball with key (1) above?

```
-----BEGIN PGP PUBLIC KEY BLOCK-----

mDMEaBgeSxYJKwYBBAHaRw8BAQdArfTkd8AZPsb68ziVc77qmbSQ0JwKT8aB4AFa
Irel4+e0Jkp1YW4gUGFibG8gTG96YW5vIDxsaWJyZWRlYkBnbWFpbC5jb20+iJkE
ExYKAEEWIQQI5IcMtubdgbCN/BJjVV9avWmX8AUCaBgeSwIbAwUJBaOagAULCQgH
AgIiAgYVCgkICwIEFgIDAQIeBwIXgAAKCRBjVV9avWmX8H/LAP9C3nekvKlb4I8b
C9FZt2a/l26eqXgiWSRS18GnQkTMrgEAsP2XXdC/YdRZsYJMjscX3L3bTpBuCBzO
93kXzVo5SAy4OARoGB5LEgorBgEEAZdVAQUBAQdAYGY2iapESTQbMjpYDhIvte4r
uxxm93kruNL5tAXCSHgDAQgHiH4EGBYKACYWIQQI5IcMtubdgbCN/BJjVV9avWmX
8AUCaBgeSwIbDAUJBaOagAAKCRBjVV9avWmX8Mn3AQC8Mz8egvLpvoX+wCqSYnJu
YP6CVVXAuIAiA4aEt5LYhgD+KuMZ2bRX+LCCvj4E9X22cpMglXW0E0NuDlsf9/WE
SA0=
=4EfG
-----END PGP PUBLIC KEY BLOCK-----
```

As far as I can see, the above is your public key for (1) but that doesn't appear to validate the asset via its published .asc file.

**Comment by libredeb** [comment]
@fossfreedom I don't know what is going wrong... is there anything I can do to validate that everything is ok?

![Image](https://github.com/user-attachments/assets/1cf6feb6-c1ab-4121-824c-42b8cda4c342)

**Comment by fossfreedom** [comment]
@libredeb hi - I was just going to go through the upload process when I realised that you have uploaded the .asc signature - but you haven't got the corresponding tarball itself to download i.e.  lightpad-0.0.10.tar.xz

**Comment by fossfreedom** [comment]
You should upload the same tarball created and signed by the mkrelease.sh or the other example script rather than depending upon what github produces as per https://github.com/libredeb/lightpad/issues/32#issuecomment-2865557667
